src/web/components/TeamMember/TeamMembersTable.spec.tsx

```diff
@@ -0,0 +1,82 @@
+import { render, screen } from '@testing-library/react';
+
+import { Participant,ParticipantDTO } from '../../../api/entities/Participant';
+import { UserRoleId } from '../../../api/entities/UserRole';
+import { UserWithCurrentParticipantRoleNames } from '../../../api/services/usersService';
+import { createMockParticipant, createMockUser } from '../../../testHelpers/dataMocks';
+import {
+  createParticipantContextValue,
+  createUserContextValueWithParticipantRoleName,
+  TestContextProvider,
+} from '../../../testHelpers/testContextProvider';
+import { UserContextWithSetter } from '../../contexts/CurrentUserProvider';
+import { ParticipantWithSetter } from '../../contexts/ParticipantProvider';
+import TeamMembersTable from './TeamMembersTable';
+
+const renderTeamMembersTableWithContext = (
+  userContextValue: UserContextWithSetter,
+  participantContextValue: ParticipantWithSetter,
+  teamMembers: UserWithCurrentParticipantRoleNames[]
+) => {
+  return render(
+    <TestContextProvider
+      userContextValue={userContextValue}
+      participantContextValue={participantContextValue}
+    >
+      <TeamMembersTable
+        teamMembers={teamMembers}
+        onAddTeamMember={() => {
+          return Promise.resolve();
+        }}
+        onRemoveTeamMember={() => {
+          return Promise.resolve();
+        }}
+        onUpdateTeamMember={() => {
+          return Promise.resolve();
+        }}
+        resendInvite={() => {
+          return Promise.resolve();
+        }}
+      />
+    </TestContextProvider>
+  );
+};
+
+describe('manage team member functionality testing', () => {
+  let participantContextValue: ParticipantWithSetter;
+  let userContextValue: UserContextWithSetter;
+  let mockParticipant: ParticipantDTO;
+  let mockUser: UserWithCurrentParticipantRoleNames;
+  beforeAll(() => {
+    mockParticipant = createMockParticipant();
+    participantContextValue = createParticipantContextValue(mockParticipant);
+    mockUser = createMockUser([mockParticipant] as Participant[]);
+  });
+
+  const teamMemberManageRoles = [
+    { id: UserRoleId.UID2Support, roleName: 'UID2Support' },
+    { id: UserRoleId.Admin, roleName: 'Admin' },
+  ];
+  it.each(teamMemberManageRoles)(
+    'should include all manage team member functionality',
+    (manageRole) => {
+      mockUser.currentParticipantUserRoles = [manageRole];
+      userContextValue = createUserContextValueWithParticipantRoleName(mockUser);
+      renderTeamMembersTableWithContext(userContextValue, participantContextValue, [mockUser]);
+      const addTeamMemberButton = screen.getByRole('button', { name: 'Add Team Member' });
+      expect(addTeamMemberButton).toBeInTheDocument();
+      const actionsHeader = screen.getByRole('columnheader', { name: 'Actions' });
+      expect(actionsHeader).toBeInTheDocument();
+      expect(screen.getByTestId('action-cell')).toBeInTheDocument();
```

Comment:
I think we can use a better RTL query here, so that we can align more closely to the user behaviour (per: https://testing-library.com/docs/queries/about/#priority). If we add `aria-label` to each of the action buttons that don't have a label, we can improve accessibility but then also use `getByLabelText()` instead of `getByTestId()`. 

e.g. For the pencil button, we can change it to:
```
<ActionButton
  onClick={onOpenChangeTeamMemberDialog}
  icon='pencil'
  aria-label='Edit Team Member'
/>
```
then test it via: 
```
expect(screen.getByLabelText('Edit Team Member')).toBeInTheDocument();
```

How's that sound?

Reply:
Fixed!